src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -687,48 +892,8 @@ TER DirectStepI::check (StrandContext const& ctx) const
         }
     }
 
-    // The previous checks apply in all cases.  The following checks apply
-    // only if we're not offer crossing because they require a preexisting
-    // trustline.
-    if (offerCrossing_)
-        return tesSUCCESS;
-
-    {
-        auto sleLine = ctx.view.read (keylet::line (src_, dst_, currency_));
-        if (!sleLine)
-        {
-            JLOG (j_.trace()) << "DirectStepI: No credit line. " << *this;
-            return terNO_LINE;
-        }
-
-        auto const authField = (src_ > dst_) ? lsfHighAuth : lsfLowAuth;
-
-        if (((*sleSrc)[sfFlags] & lsfRequireAuth) &&
-            !((*sleLine)[sfFlags] & authField) &&
-            (*sleLine)[sfBalance] == zero)
-        {
-            JLOG (j_.warn())
-                << "DirectStepI: can't receive IOUs from issuer without auth."
-                << " src: " << src_;
-            return terNO_AUTH;
-        }
-    }
-
-    {
-        auto const owed = creditBalance (ctx.view, dst_, src_, currency_);
-        if (owed <= zero)
-        {
-            auto const limit = creditLimit (ctx.view, dst_, src_, currency_);
-            if (-owed >= limit)
-            {
-                JLOG (j_.debug())
-                    << "DirectStepI: dry: owed: " << owed << " limit: " << limit;
-                return tecPATH_DRY;
-            }
-        }
-    }
-
-    return tesSUCCESS;
+    // Perform any checks that are specific to the Policy.
```

Comment:
Nit: This is commenting obvious code. I'm making a note because I noticed a couple comments like this in the PR. No change needed for this PR, but I would raise your threshold for what to comment one notch.